### PR TITLE
GNZ-1336: Add handler generator for Genezio Runtime

### PR DESCRIPTION
**Pull request by valighita**
## Type of change

-   [x] 🍕 New feature

## Description

When using the `capybara` cloud provider, generate the node handler differently as required by the genezio runtime.

### CODE REVIEW

File: src/bundlers/node/nodeJsBundler.ts [hunk 1]
```diff
@@ -377,11 +379,24 @@ export class NodeJsBundler implements BundlerInterface {
...
+        if (cloudProvider === CloudProviderIdentifier.GENEZIO) {
+            handlerGenerator = lambdaHandlerGenerator;
+        } else if (
+            cloudProvider === CloudProviderIdentifier.CAPYBARA ||
+            cloudProvider === CloudProviderIdentifier.CAPYBARA_LINUX
+        ) {
+            handlerGenerator = genezioRuntimeHandlerGenerator;
+        } else {
+            throw new Error(`Can't generate handler for cloud provider ${cloudProvider}.`);
+        }
```
Comment (costinsin) [inline src/bundlers/node/nodeJsBundler.ts:398]: I would use a switch here
```suggestion
        switch(cloudProvider) {
            case CloudProviderIdentifier.GENEZIO:
                 handlerGenerator = lambdaHandlerGenerator;
            case CloudProviderIdentifier.CAPYBARA:
            case CloudProviderIdentifier.CAPYBARA_LINUX:
                 handlerGenerator = genezioRuntimeHandlerGenerator;
            default:
                 throw new Error(`Can't generate handler for cloud provider ${cloudProvider}.`);
        }
```

**Review by costinsin — APPROVED** [review]

**Review by MagoDelBlocco — APPROVED** [review]
LGTM

File: src/bundlers/node/genezioRuntimeHandlerGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
+/* eslint-disable no-useless-escape */
+
+export const genezioRuntimeHandlerGenerator = (className: string): string => `
+/** This is an auto generated code. This code should not be modified since the file can be overwritten
+ *  if new genezio commands are executed.
+ */
+
+import {  ${className.replace(/["]/g, "")} as genezioClass } from "./module.mjs";
```
Comment (vladiulianbogdan) [inline src/bundlers/node/genezioRuntimeHandlerGenerator.ts:8]: why is this necessary?

File: src/bundlers/node/nodeJsBundler.ts [hunk 3]
```diff
@@ -377,11 +379,24 @@ export class NodeJsBundler implements BundlerInterface {
         const mode = input.extra.mode;
         const tmpFolder = input.extra.tmpFolder;
         const cwd = input.projectConfiguration.workspace?.backend || process.cwd();
+        const cloudProvider = input.projectConfiguration.cloudProvider;
 
         if (mode === "development" && !tmpFolder) {
             throw new Error("tmpFolder is required in development mode.");
         }
 
+        var handlerGenerator: (className: string) => string;
```
Comment (vladiulianbogdan) [inline src/bundlers/node/nodeJsBundler.ts:388]: I would move this whole logic in another function in this file

**Review by vladiulianbogdan — APPROVED** [review]

File: src/bundlers/node/genezioRuntimeHandlerGenerator.ts (hunk 2 above)
Comment (valighita) [reply src/bundlers/node/genezioRuntimeHandlerGenerator.ts:8]: I copied this from the lambda generator. @radu1122 @costinsin any input?
